src/sequelize_connection_manager.ts

```diff
@@ -0,0 +1,118 @@
+import * as bcrypt from 'bcryptjs';
+import * as fsExtra from 'fs-extra';
+import {Logger} from 'loggerhythm';
+import * as path from 'path';
+import * as Sequelize from 'sequelize';
+
+const logger: Logger = Logger.createLogger('essential-projects:sequelize_connection_manager');
+
+/**
+ * Creates, manages and destroys connections to Sequelize based databases.
+ */
+export class SequelizeConnectionManager {
+
+  private connections: {[hash: string]: Sequelize.Sequelize} = {};
+
+  /**
+   * Returns a sequelize connection for the given configuration.
+   *
+   * @async
+   * @param  config          Contains the settings with which to establish a
+   *                         database connection.
+   *                         Required parameters are host, port, database,
+   *                         dialect, username and password.
+   * @param  config.host     The name of the host where the database is located.
+   * @param  config.port     The port by which to connect to the host.
+   * @param  config.dialect  The type of database to which to connect
+   *                         (postgres, sqlite, mysql, etc).
+   * @param  config.database The name of the database to connect to.
+   * @param  config.username The username with which to connect to the database.
+   * @param  config.password The password with which to connect to the database.
+   * @return                 The connection for the passed configuration.
+   */
+  public async getConnection(config: Sequelize.Options): Promise<Sequelize.Sequelize> {
+
+    let hash: string;
+
+    const dbToUse: string = config.dialect === 'sqlite'
+      ? config.storage
+      : config.database;
+
+    if (config.dialect === 'sqlite') {
+      hash = this._getHash(config.storage, config.username, config.password);
+    } else {
+      hash = this._getHash(config.database, config.username, config.password);
+    }
+
+    const connectionExists: boolean = this.connections[hash] !== undefined;
+    if (connectionExists) {
+      logger.info(`Active connection to '${dbToUse}' found.`);
+
+      return Promise.resolve(this.connections[hash]);
+    }
+
+    if (config.dialect === 'sqlite') {
+      const isAbsolutePath: boolean = path.isAbsolute(config.storage);
```

Comment:
```suggestion
      const pathIsAbsolute: boolean = path.isAbsolute(config.storage);
```

I would prefer this; It moves the important bit to the front, namely: What is this about? Path.

This is the first thing to know: Context (the Path of something) then comes the condition.